resources/cleanup_scripts/deduplicate.py

```diff
@@ -0,0 +1,300 @@
+
+
+def _resolve_wipac_location_filepath(fc_meta: FCMetadata) -> FCMetadata:
+    # replace WIPAC-site path (these will differ b/c they are the logical_name)
+    for i in range(len(fc_meta["locations"])):  # pylint: disable=C0200
+        # allow in-line changes
+        if fc_meta["locations"][i]["site"] == "WIPAC":
+            path = _get_good_path(fc_meta["locations"][i]["path"])
+            fc_meta["locations"][i]["path"] = path
+    return fc_meta
+
+
+def _resolve_season_value(fc_meta: FCMetadata) -> FCMetadata:
+    if "offline_processing_metadata" in fc_meta:
+        season = fc_meta["offline_processing_metadata"]["season"]
+        fc_meta["offline_processing_metadata"]["season"] = int(season)
+    return fc_meta
+
+
+def has_good_twin(rc: RestClient, evil_twin: FCMetadata) -> bool:
+    """Return whether the `evil_twin` has a good twin.
+
+    If the two sets of metadata are not compatible, raise an Exception.
+    """
+    try:
+        good_twin = _find_fc_metadata(rc, _get_good_path(evil_twin["logical_name"]))
+    except FileNotFoundError:
+        return False
+
+    # resolve special fields
+    evil_twin = _resolve_deprecated_fields(evil_twin)
+    evil_twin = _resolve_gcd_filepath(evil_twin)
+    evil_twin = _resolve_wipac_location_filepath(evil_twin)
+    #
+    evil_twin = _resolve_season_value(evil_twin)
+    good_twin = _resolve_season_value(good_twin)
+
+    # compare metadata
+    try:
+        # compare "locations"-fields
+        if not _compatible_locations_values(
+            evil_twin["locations"], good_twin["locations"]
+        ):
+            raise Exception("Locations metadata not compatible")
+
+        # compare "meta_modify_date"-fields
+        if _evil_twin_was_updated_later(evil_twin, good_twin):
+            raise Exception("Evil twin was updated after the good twin")
+
+        # compare basic fields
+        ignored_fields = [
+            "_links",
+            "logical_name",
+            "uuid",
+            "locations",
+            "meta_modify_date",
+        ]
+        if not _compare_twins(evil_twin, good_twin, ignored_fields):
+            raise Exception(f"Fields don't match (disregarding: {ignored_fields})")
+
+    except Exception:
+        logging.critical(f"evil_twin={evil_twin}")
+        logging.critical(f"good_twin={good_twin}")
+        raise
+
+    return True
+
+
+def bad_fc_metadata(rc: RestClient) -> Generator[FCMetadata, None, None]:
+    """Yield each FC entry (w/ full metadata) rooted at /mnt/lfs*/.
+
+    Search will be halted either by a REST error, manually by the user,
+    or when the FC has been exhausted.
+    """
+    previous_page: List[Dict[str, Any]] = []
+    page = 0
+    while True:
+        logging.info(
+            f"Looking for more bad-rooted paths (page={page}, limit={PAGE_SIZE})..."
+        )
+
+        # Query
+        body = {"start": page * PAGE_SIZE, "limit": PAGE_SIZE, "all-keys": True}
+        resp = rc.request_seq("GET", "/api/files", body)
```

Comment:
Would it be faster to do the logical name query here, so it happens on the server side?  There should be a way to send a mongo query to the server to ask for only prefixed logical names.

Reply:
Is that possible?

Reply:
```python
rc.request_seq('GET', '/api/files', {'limit': 1, 'query':json.dumps({"logical_name":{ '$regex': '^\/mnt\/lfs.*' }})})
```

Reply:
I didn't know mongo supported regex--that cleans things up nicely. 

Now I'm thinking that adding a "directory" base argument (which would essentially make a `$regex` query on the path name) would be useful.

Reply:
Sounds like another PR :smile:

Reply:
https://github.com/WIPACrepo/file_catalog/issues/77